packages/affine/block-surface/src/element-model/mindmap.ts

```diff
@@ -885,9 +885,18 @@ export class MindmapElementModel extends GfxGroupLikeElementModel<MindmapElement
       removedDescendants.forEach(id => surface.removeElement(id));
     });
 
+    // Since the root node is removed, the tree is invalid
+    // this happens on the entire mindmap is deleting
+    if (!this._tree || id === this._tree.id) {
+      // @ts-ignore
+      this._tree = null;
```

Comment:
I think you could just modify the `buildTree` method by checking if the children is empty or undefined.